vortex-array/src/array/chunked/canonical.rs

```diff
@@ -47,9 +47,17 @@ pub(crate) fn try_canonicalize_chunks(
         // Extension arrays wrap an internal storage array, which can hold a ChunkedArray until
         // it is safe to unpack them.
         DType::Extension(ext_dtype, _) => {
+            if chunks.is_empty() {
```

Comment:
there's a fun case of "What is the canonical storage type for an extension dtype"?